code/modules/client/preference_setup/general/03_body.dm

```diff
@@ -517,7 +523,30 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 
 			valid_hairstyles[hairstyle] = GLOB.hair_styles_list[hairstyle]
 
-		var/new_h_style = tgui_input_list(user, "Choose your character's hair style.", "Character Preference", valid_hairstyles, pref.h_style)
+		var/new_h_style
+		var/selected_style_index = valid_hairstyles.Find(pref.h_style)
+
+		if (href_list["next_hair_style"])
+			if (selected_style_index >= valid_hairstyles.len)
+				to_chat(user, SPAN_NOTICE("You're at the bottom of the list! Delightful!"))
```

Comment:
These should just loop around

Reply:
Something like `1, 2, 3, 1, 2, 3`? When reaching the end of the list, just change the index to the very first one?